src/functional/ffront/past_to_itir.py

```diff
@@ -95,9 +95,14 @@ def _gen_size_params_from_program(self, node: past.Program):
         """Generate symbols for each field param and dimension."""
         size_params = []
         for param in node.params:
-            if isinstance(param.type, common_types.FieldType):
-                for dim_idx in range(0, len(param.type.dims)):
+            if type_info.is_field_type_or_tuple_of_field_type(param.type):
+                fields_dims: list[list[Dimension]] = (
+                    type_info.primitive_constituents(param.type).getattr("dims").to_list()
+                )
+                assert all(field_dims == fields_dims[0] for field_dims in fields_dims)
```

Comment:
```suggestion
                assert all(field_dim == fields_dims[0] for field_dim in fields_dims)
```

Reply:
`field_dims` is a list of dimensions (see the `list[list[Dimension]]` annotation above), hence the name. Contrary to that _field**s**\_dims_ is a list of dimensions of all the field**s**.